accelerator/proxy_apis.c

```diff
@@ -313,46 +321,52 @@ status_t api_get_trytes(const iota_client_service_t* const service, const char*
   return ret;
 }
 
-status_t api_remove_neighbors(const iota_client_service_t* const service, const char* const obj, char** json_result) {
+status_t api_proxy_apis(const iota_client_service_t* const service, const char* const obj, char** json_result) {
   status_t ret = SC_OK;
-  remove_neighbors_req_t* req = remove_neighbors_req_new();
-  remove_neighbors_res_t* res = remove_neighbors_res_new();
-  char_buffer_t* res_buff = char_buffer_new();
-  if (req == NULL || res == NULL || res_buff == NULL) {
-    ret = SC_TA_OOM;
-    ta_log_error("%s\n", "SC_TA_OOM");
-    goto done;
-  }
-
-  lock_handle_lock(&cjson_lock);
-  if (service->serializer.vtable.remove_neighbors_deserialize_request(obj, req) != RC_OK) {
-    lock_handle_unlock(&cjson_lock);
-    ret = SC_CCLIENT_JSON_PARSE;
-    ta_log_error("%s\n", "SC_CCLIENT_JSON_PARSE");
-    goto done;
-  }
-  lock_handle_unlock(&cjson_lock);
-
-  lock_handle_lock(&cjson_lock);
-  if (iota_client_remove_neighbors(service, req, res) != RC_OK) {
-    lock_handle_unlock(&cjson_lock);
-    ret = SC_CCLIENT_FAILED_RESPONSE;
-    ta_log_error("%s\n", "SC_CCLIENT_FAILED_RESPONSE");
-    goto done;
+  const uint8_t max_cmd_len = 30;
+  char command[max_cmd_len];
+
+  proxy_apis_command_req_deserialize(obj, command);
+  // TODO more proxy APIs should be implemented
+  enum {
+    H_checkConsistency = 3512812181U,
+    H_findTransactions = 1314581663U,
+    H_getBalances = 185959358U,
+    H_getInclusionStates = 3627925933U,
+    H_getNodeInfo = 1097192119U,
+    H_getTrytes = 3016694096U
+  };
+
+  switch (hash_algo_djb2(command)) {
+    case H_checkConsistency:
+      ret = api_check_consistency(service, obj, json_result);
+      break;
+    case H_findTransactions:
+      ret = api_find_transactions(service, obj, json_result);
+      break;
+    case H_getBalances:
+      ret = api_get_balances(service, obj, json_result);
+      break;
+    case H_getInclusionStates:
+      ret = api_get_inclusion_states(service, obj, json_result);
+      break;
+    case H_getNodeInfo:
+      ret = api_get_node_info(service, json_result);
+      break;
+    case H_getTrytes:
+      ret = api_get_trytes(service, obj, json_result);
+      break;
+
+    default:
+      ret = SC_HTTP_URL_NOT_MATCH;
+      ta_log_error("%s\n", "SC_HTTP_URL_NOT_MATCH");
+      goto done;
```

Comment:
Would it be better if the statement `goto done` is replaced with `return ret`?

Reply:
sure